### ISSUE TITLE
Exception in org.jdbi.v3.core.statement.StatementExceptions

### ISSUE THREAD

**Issue by vaclavhodek** [jdbi/jdbi#1640]
I have the query that is shown below. The app is written in Kotlin and the dialect is PostgreSQL. There is a mistake in the query - I bind list as `hashIds` but accidentaly used `<hashId>`.

```
@SqlQuery(
"""
SELECT prjs.project_id AS project_id, phrs.phrase_id AS phrase_id, phrs.hash_id AS hash_id, lngs.lang_id AS lang_id FROM 
    (SELECT phrase_id, hash_id FROM pmatch_links WHERE hash_id IN (<hashId>) GROUP BY phrase_id, hash_id) AS phrs 
CROSS JOIN 
    (SELECT DISTINCT lang_id FROM pmatch_translations WHERE hash_id IN (<hashId>)) AS lngs
INNER JOIN 
    pmatch_projects AS prjs ON phrs.phrase_id = prjs.phrase_id
WHERE lngs.lang_id NOT IN 
    (SELECT lang_id FROM pmatch_links AS pl WHERE pl.phrase_id = phrs.phrase_id)
"""
)
fun getPhraseMatchSets(
    @BindList("hashIds") hashIds: List<Long>
): List<PhraseMatchSet>
```

Instead of meaningful exception the whole app crashes as `e.printStackTrace()` in this snippet:

```
try {
    dao.getPhraseMatchSets(hashIds)
} catch (e: Exception) {
    e.printStackTrace()
}
```

produces exception: 

```
java.lang.NullPointerException
	at org.jdbi.v3.core.statement.StatementExceptions.limit(StatementExceptions.java:140)
	at org.jdbi.v3.core.statement.StatementExceptions$MessageRendering$3.render(StatementExceptions.java:106)
	at org.jdbi.v3.core.statement.StatementExceptions$MessageRendering.apply(StatementExceptions.java:132)
	at org.jdbi.v3.core.statement.StatementException.getMessage(StatementException.java:72)
	at java.lang.Throwable.getLocalizedMessage(Throwable.java:391)
	at java.lang.Throwable.toString(Throwable.java:480)
	at java.lang.String.valueOf(String.java:2994)
	at java.io.PrintStream.println(PrintStream.java:821)
	at java.lang.Throwable$WrappedPrintStream.println(Throwable.java:748)
	at java.lang.Throwable.printStackTrace(Throwable.java:655)
	at java.lang.Throwable.printStackTrace(Throwable.java:643)
	at java.lang.Throwable.printStackTrace(Throwable.java:634)
	at com.localazy.pmatch.processors.InsertProcessor.insert(InsertProcessor.kt:244)
        ...
```

It took me a while to figure out the problem but once I've changed `<hashId>` to `<hashIds>` in the query shown above, everything works like expected. 

If I don't call `printStackTrace()` and `getMessage()` on the exception, I can still access stacktrace and trace the problem to `org.jdbi.v3.core.statement.DefinedAttributeTemplateEngine:render; line 70`. 

I'm currently running JDBI 3.9.0 as we haven't tested other projects with the latest version yet.

**Comment by stevenschlansker** [comment]
Hi,
I believe this has already been fixed since 3.9.0.  With current release, the error message looks like:

```
org.jdbi.v3.core.statement.UnableToCreateStatementException: Error rendering SQL template: 'select (<hashIds>)' [statement:"null", arguments:{positional:{}, named:{}, finder:[]}]
	at org.jdbi.v3.core.statement.DefinedAttributeTemplateEngine.lambda$4(DefinedAttributeTemplateEngine.java:99)
...
Caused by: org.jdbi.v3.core.statement.UnableToCreateStatementException: Undefined attribute for token '<hashIds>' [statement:"null", arguments:{positional:{}, named:{}, finder:[]}]
	at org.jdbi.v3.core.statement.DefinedAttributeTemplateEngine.lambda$3(DefinedAttributeTemplateEngine.java:79)
	... 32 more
```

Hopefully this error message is clearer?  Let us know if there's any other improvements to be made here.  Thanks for reporting.

**Comment by vaclavhodek** [comment]
I just tried 3.12.0 and I can confirm that the problem is fixed. Thanks a lot!
